index.src.html

```diff
@@ -864,22 +861,18 @@ <h3 id="determine-requests-referrer" dfn export>
         <dd>
           <ol>
             <li>
-              If <var>request</var> is a <a>same-origin request</a>, then
-              return <var>referrerURL</var>.
+              If |referrerURL| is <code>no referrer</code>, then return <code>no referrer</code>.
             </li>
             <li>
-              If |environment| is not null:
-              <ol>
-                <li>
-                  If |environment|'s <a for="environment settings object">HTTPS state</a> is "`modern`" <em>and</em>
-                  <var>request</var>'s
-                  <a href="https://fetch.spec.whatwg.org/#concept-request-current-url">current
-                  URL</a> is not a <a>potentially trustworthy URL</a>
-                  <ol>
-                      <li>Return <code>no referrer</code>.
-                  </ol>
-                </li>
-              </ol>
+              If the  <a for=url>origin</a> of |referrerURL| and the <a for=url>origin</a> of
+              <var>request</var>'s <a for=request>current url</a> are <a lt="same origin">the
+              same</a>, then return |referrerURL|.
```

Comment:
These look good, but I wonder if it would be cleaner to pull these comparisons out to a variable `same origin with referrer` and refer to it, since we do the comparisons several times.

Reply:
yes, also looks good to me.